transport/http/inbound.go

```diff
@@ -55,9 +55,10 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 // sharing this transport.
 func (t *Transport) NewInbound(addr string, opts ...InboundOption) *Inbound {
 	i := &Inbound{
-		once:   sync.Once(),
-		addr:   addr,
-		tracer: t.tracer,
+		once:      sync.Once(),
+		addr:      addr,
+		tracer:    t.tracer,
+		transport: t,
```

Comment:
No need to change this: id love if we changed as a team to unkeyed fields, it ends up catching a lot more at compile time, at minimal cost